arch/arm64/src/qemu/qemu_boot.c

```diff
@@ -167,6 +171,10 @@ void arm64_chip_boot(void)
 
   arm64_mmu_init(true);
 
+#ifdef CONFIG_DEVICE_TREE
+  fdt_register((FAR const char *)0x40000000);
```

Comment:
@xiaoxiang781216 @leo11261 this should not be hard coded and instead extracted from the register as I did for rv.  

See my comments here for arm and arm64 

https://github.com/apache/nuttx/pull/9204#issuecomment-1550762515

Reply:
but from https://www.qemu.org/docs/master/system/arm/virt.html?highlight=device%20tree%20blob, only -kernel with non-ELF file could get the fdt location from r2 or x0.